File: Find-The-Least-Frequent-Digit.java
1class Solution {
2    public int getLeastFrequentDigit(int n) {
3        int[] freq = new int[10];
4        Arrays.fill(freq,0);
5
6        while(n>0){
7            freq[n%10]++;
8            n/=10;
9        }
10        int mindig = Integer.MAX_VALUE, minfreq = Integer.MAX_VALUE;
11        for(int digit = 0 ; digit <= 9 ; digit++){
12            if(freq[digit] < minfreq && freq[digit] > 0){
13                minfreq = freq[digit];
14                mindig = digit;
15            }
16        }
17        return mindig;
18    }
19}

File: Find-The-Least-Frequent-Digit.cpp
1class Solution {
2public:
3    int getLeastFrequentDigit(int n) {
4        int freq[10] = {0};
5        while(n>0){
6            freq[n%10]++;
7            n/=10;
8        }
9        int mindigit = INT_MAX,minfreq = INT_MAX;
10        for(int digit = 0 ; digit < 10 ; digit++){
11            if(freq[digit] < minfreq && freq[digit]!=0){
12                minfreq = freq[digit];
13                mindigit = digit;
14            }
15        }
16        return mindigit;
17    }
18};